source/common/tcp/conn_pool.h

```diff
@@ -0,0 +1,203 @@
+#pragma once
+
+#include <list>
+#include <memory>
+
+#include "envoy/event/deferred_deletable.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/connection.h"
+#include "envoy/network/filter.h"
+#include "envoy/stats/timespan.h"
+#include "envoy/tcp/conn_pool.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/common/linked_object.h"
+#include "common/common/logger.h"
+#include "common/http/conn_pool_base.h"
+#include "common/network/filter_impl.h"
+
+namespace Envoy {
+namespace Tcp {
+
+class ConnPoolImpl;
+
+class TcpPendingRequest : public Envoy::ConnectionPool::PendingRequest {
+public:
+  TcpPendingRequest(Envoy::ConnectionPool::ConnPoolImplBase& parent,
+                    Tcp::ConnectionPool::Callbacks& callbacks)
+      : Envoy::ConnectionPool::PendingRequest(parent), callbacks_(callbacks) {}
+  void* context() override { return static_cast<void*>(&callbacks_); }
```

Comment:
I'm not a fan of the `void*`, but I assume that's part of the interface?

Reply:
Yeah, potential workaround in https://github.com/envoyproxy/envoy/pull/11796